src/org/labkey/test/util/TestDataGenerator.java

```diff
@@ -642,7 +644,22 @@ public static String randomFieldName(@NotNull String part, @Nullable Integer num
         return randomFieldName;
     }
 
-    private static boolean isDomainAndFieldNameInvalid(DomainUtils.DomainKind domainKind, @Nullable String domainName, @Nullable String fieldName)
+    private static boolean isDomainAndFieldNameInvalid(DomainKind domainKind, @Nullable String domainName, @Nullable String fieldName)
+    {
+        if (fieldName != null && fieldName.length() > 64 && fieldName.toLowerCase().contains("key")) // Not guaranteed but likely a list key
+            return true; // Issue 53706: List key field name length is limited to 64 characters
```

Comment:
As @labkey-jeckels mentioned, this issue will be fixed in develop very soon, so this code can be removed there.